test/export/test_export.py

```diff
@@ -1447,23 +1446,6 @@ def f(x):
         inp = torch.randn(2)
         self.assertTrue(torch.allclose(ep(inp), torch.nonzero(inp)))
 
-    def test_redundant_asserts(self):
-        def f(x):
-            y = x.item()
-            torch._constrain_as_size(y)
-            return torch.zeros(y)
-
-        ep = export(f, (torch.tensor([3]),))
-        self.assertExpectedInline(str(ep.graph_module.code).strip(), """\
-def forward(self, arg0_1):
-    _local_scalar_dense = torch.ops.aten._local_scalar_dense.default(arg0_1);  arg0_1 = None
-    ge = _local_scalar_dense >= 0
-    scalar_tensor = torch.ops.aten.scalar_tensor.default(ge);  ge = None
-    _assert_async = torch.ops.aten._assert_async.msg(scalar_tensor, '_local_scalar_dense is outside of inline constraint [0, inf].');  scalar_tensor = None
```

Comment:
Hmm I just realized removing assertions could lead to following behaviour:
If user wants to just remove assertion and still have the working graph, it wouldn't work now because they will have to delete constraint_as_size call. If you delete that, they cannot retrace the graph anymore.

Reply:
Hahah you are right. I feel it might be too strict tho.

Reply:
The assertions already exist in the constrain_as_size call. My diff here just removes the use of the pass -- maybe I'm misunderstanding something.